python_modules/dagster/dagster/_core/ext/context.py

```diff
@@ -38,12 +40,17 @@ def handle_message(self, message: ExtMessage) -> None:
         elif message["method"] == "log":
             self._handle_log(**message["params"])  # type: ignore
 
-    def _handle_report_asset_metadata(self, asset_key: str, label: str, value: Any) -> None:
+    def _handle_report_asset_metadata(
+        self, asset_key: str, label: str, value: Any, type: ExtMetadataType  # noqa: A002
+    ) -> None:
         check.str_param(asset_key, "asset_key")
         check.str_param(label, "label")
+        check.opt_literal_param(type, "type", get_args(ExtMetadataType))
         key = AssetKey.from_user_string(asset_key)
         output_name = self._context.output_for_asset_key(key)
-        self._context.add_output_metadata({label: value}, output_name)
+        # metadata type values correspond to static methods on MetadataValue
```

Comment:
we should not be relying on keeping static methods on `MetadataValue` up-to-date and the members of ExtMetadataType synced via convention alone. Needs to be guaranteed by some sort of code artifact or a unit test that ensures that they correspond by using reflection apis